### ISSUE TITLE
Unable to publish into a TFS repository

### ISSUE THREAD

**Issue by wwu206** [archimatetool/archi-modelrepository-plugin#90]
I' having trouble to publish models to our TFS git reposirory.
The error is:
Unable to tunnel through proxy. Proxy returns "HTTP/1.1 403 Forbiden"

Already tried with and without proxy. With/without login user.

Any ideia or help to solve this problem was appreciated.

Rui

**Comment by Phillipus** [comment]
Amy more details than that? What's the format of the URL? HTTPS? SSH?

**Comment by wwu206** [comment]
Hi Phil.

Thanks for your quick answer.

The URL is this one with HTTPS:
[cid:image002.png@01D4E314.F0C2DB20]

[cid:image003.png@01D4E314.F0C2DB20]

Any clues?

What am I doing wrong?

Best regards,
Rui Nuno Simões
Arquiteto de SI/TI | IS/IT Architect
[Assinatura C]
Departamento de Sistemas e Tecnologias de Informação | Information Systems and Technology Department
Núcleo de Arquitetura de SI/TI | IS/IT Architecture Unit

Rua Francisco Ribeiro, 2 | 1150-165 Lisboa
T +351 213130762
F +351 213143873
rnsimoes@bportugal.pt<mailto:rnsimoes@bportugal.pt>| www.bportugal.pt<http://www.bportugal.pt/pt-PT/Paginas/inicio.aspx>
P Antes de imprimir, pense no ambiente
       Before printing, think about the environment

**Comment by Phillipus** [comment]
Hard to say. Have a look at Issue #66 and try setting "jdk.http.auth.proxying.disabledSchemes="

**Comment by wwu206** [comment]
Already tried issue #66 but no change

**Comment by Phillipus** [comment]
Does the error dialog box start with these words "Cannot connect to proxy:"?

**Comment by richteambs** [comment]
This looks like it might be the same problem I'm having. When I click Collaboration - Pubilsh Changes, I get an error dialog with a "not authorized" message.

Publishing used to work until recently. Unfortunately, several things have changed since I last know it worked: 

1. I updated the plugin to the latest version.
2. The TFS server was updated to Azure DevOps 2019 (the latest rebrand)
3. My password has changed

I can't tell which of these has caused it to break - although I suspect not 3, because I've entered the new password in the authentication dialog.

**Comment by theazyfa** [comment]
Hello, 
I've the same issue.
I add my config maybe it can help.
We have a **TFS** defined on **HTTP** (_On-Prem version Version 16.131.27701.1_)
Archimate _4.4.0_
Plugin _0.5.1_

I start by **adding the model to workspace and publishing it**
In the popup it say "Publishing..."
And an error occurs ; 
"
There was an error : http://tfs****:8080/tfs/COLLECTION/_git/Archimate: not authorized
"

I can't find if it's a login/password issue or not.
In my case, our login are like this : Domain\login or login@domain (upn or samaccountname)
Is it okay to have special character in the login? (\ or @)

**Comment by Phillipus** [comment]
Just to say that I personally can't debug this problem unless I have a similar setup and access to TFS. So we really need someone who can run the code and debug it in this same kind of scenario.

**Comment by theazyfa** [comment]
Hi, 
I config a devel env on my pc ( eclipse oxigen + archimate code + plugin code) to try to debug my issue.
And Finally I find an issue in my TFS config.
We need to add & enable "**Basic Authentication**" on _TFS Website_ in _IIS Config_ and also on the _tfs_ virtual directory defined.

[config IIS](https://docs.microsoft.com/en-us/iis/configuration/system.webserver/security/authentication/basicauthentication)

I find it with this [SF response ](https://stackoverflow.com/questions/22939372/how-do-i-enable-tee-eclipse-plugin-on-linux-to-clone-a-git-repository-in-tfs-201)

For me, there is NO bug in the plugin.
Sorry for the mistake (and for my english)
Mike.

**Comment by Phillipus** [comment]
@theazyfa Good news!

**Comment by richteambs** [comment]
I just wanted to add that, although @theazyfa 's suggestion didn't fix my problem, it led to a conversation with our TFS team that did resolve it. Just in case anyone else ends up here with the same issue, here's what happened.

As part of the upgrade from TFS to Azure DevOps, the TFS team disabled Basic authentication (they only now support Windows authentication). I therefore switched from using my username / password to using [personal access tokens](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate). I needed to create a token with the following permissions:

- Code (read and write)
- Code (status)

I then pasted the PAT into the password field and left a dummy value in the username field - problem solved!
